res/controllers/Numark-NS6II-scripts.js

```diff
@@ -0,0 +1,1479 @@
+        max: 0x7F,
+        off: 0x00,
+        outValueScale: function(playpos) {
+            // check if track is loaded because playpos value is 0.5 when there isn't a track loaded.
+            return deckInfoCache.trackLoaded ?
+                Math.round(playpos * this.max) :
+                this.off;
+        },
+        send: NS6II.createIdempotentSend(NS6II.DisplayElement.prototype.send),
+    });
+
+    this.vinylStickerPositionUI = new NS6II.DisplayElement({
+        midi: [0xB0 + channelOffset, 0x06],
+        outKey: "playposition",
+        max: 0x7F,
+        off: 0x00,
+        outValueScale: function(playpos) {
+            const elapsedTime = deckInfoCache.duration * playpos;
+            return script.posMod(elapsedTime * deckInfoCache.vinylControlSpeedTypeRatio, 1) * this.max;
+        },
+        send: NS6II.createIdempotentSend(NS6II.DisplayElement.prototype.send),
+    });
+
+    this.deckLoadedConnection = engine.makeConnection(deck, "track_loaded", function(value) {
+        deckInfoCache.trackLoaded = value;
+    });
+    this.deckLoadedConnection.trigger();
+
+};
+
+NS6II.Display.prototype = new components.ComponentContainer();
+
+NS6II.PadMode = function(channelOffset) {
+    components.ComponentContainer.call(this, {});
+
+    this.constructPads = constructPad => {
+        this.pads = this.pads.map((_, padIndex) => constructPad(padIndex));
+    };
+    const makeParameterPressHandler = (control, onButtonDown) =>
+        new components.Button({
+            midi: [0x90 + channelOffset, control],
+            // never outconnect, as these buttons don't have LEDs
+            outConnect: false,
+            input: function(channelmidi, control, value, status, group) {
+                if (this.isPress(channelmidi, control, value, status)) {
+                    onButtonDown(channelmidi, control, value, status, group);
+                }
+            },
+        });
+    this.assignParameterPressHandlerLeft = onButtonDown => {
+        this.parameterLeft = makeParameterPressHandler(0x28, onButtonDown);
+    };
+    this.assignParameterPressHandlerRight = onButtonDown => {
+        this.parameterRight = makeParameterPressHandler(0x29, onButtonDown);
+    };
+    // this is a workaround for components, forEachComponent only iterates
+    // over ownProperties, so these have to constructed by the constructor here
+    // instead of being merged by the ComponentContainer constructor
+    this.pads = Array(8).fill(undefined);
+    const doNothing = () => {};
+    this.assignParameterPressHandlerLeft(doNothing);
+    this.assignParameterPressHandlerLeft(doNothing);
+};
+
+NS6II.PadMode.prototype = new components.ComponentContainer();
+
+NS6II.Pad = function(options) {
+    components.Button.call(this, options);
+};
+NS6II.Pad.prototype = new components.Button({
+    // grey could be an alternative as well as a backlight color.
+    off: NS6II.USE_BUTTON_BACKLIGHT ? NS6II.PAD_COLORS.RED.DIMMER : NS6II.PAD_COLORS.OFF,
```

Comment:
Could it be worth adding a dynamic setting for that?

Reply:
yes, I'll add that as a follow up, but I wanted to keep this branch targetted at 2.4